lib/kubeclient_next/client/rest_client.rb

```diff
@@ -35,7 +35,12 @@ def get_namespaces
 
       # Core/v1 resources are in `/api/#{version}`. In general, all other resources are found in `/apis/GROUP/VERSION`
       def endpoint
-        @endpoint ||= group ? URI.join(host, @group, @version) : URI.join(host, "/api/#{version}")
+        @endpoint ||= if group == "core"
+          URI.join(host,
+            "/api/#{version}")
+        else
+          URI.join(host, "/apis/#{@group}/#{@version}")
+        end
```

Comment:
RESTClient really shouldn't care about deriving endpoints like this, they should be provided. This way, we can query arbitrary parts of the API server, should we desire.